spec/registrant_spec.rb

```diff
@@ -0,0 +1,119 @@
+require 'spec_helper'
+require './lib/registrant'
+
+
+RSpec.describe Registrant do
+  before :each do
+    @registrant = Registrant.new(@name, @age)
+    @registrant_1 = Registrant.new('Bruce', 18, true )
+    @registrant_2 = Registrant.new('Penny', 15 )
+  end
+
+  describe '#initialize' do
+    it 'can initialize' do
+      expect(@registrant).to be_an_instance_of(Registrant)
+      expect(@registrant_1.name).to eq("Bruce")
+      expect(@registrant_1.age).to eq(18)
+      expect(@registrant_1.license_data).to eq({:written=>false, :license=>false, :renewed=>false})
+      expect(@registrant_2.name).to eq("Penny")
+      expect(@registrant_2.age).to eq(15)
+      expect(@registrant_2.license_data).to eq({:written=>false, :license=>false, :renewed=>false})
+    end
+
+    it 'has a permit' do
+      expect(@registrant_1.permit?).to be(true)
+      expect(@registrant_2.permit?).to eq(false)
+    end
+
+    it 'earns a permit' do
+      @registrant_2.earn_permit
+      expect(@registrant_2.permit?).to eq(true)
+    end
+  end
+end
+
+RSpec.describe 'getting a drivers license' do
+    before :each do
+      @registrant_1 = Registrant.new('Bruce', 18, true )
+      @registrant_2 = Registrant.new('Penny', 16 )
+      @registrant_3 = Registrant.new('Tucker', 15 )
+      @facility_1 = Facility.new({name: 'DMV Tremont Branch', address: '2855 Tremont Place Suite 118 Denver CO 80205', phone: '(720) 865-4600'})
+      @facility_2 = Facility.new({name: 'DMV Northeast Branch', address: '4685 Peoria Street Suite 101 Denver CO 80239', phone: '(720) 865-4600'})
+    end
+
+  describe 'written test' do
+    it 'can license a driver' do
+      expect(@registrant_1.license_data).to eq({:written=>false, :license=>false, :renewed=>false})
+      expect(@registrant_1.permit?).to eq(true)
+      expect(@facility_1.administer_written_test(@registrant_1)).to eq(false)
+      expect(@registrant_1.license_data).to eq({:written=>false, :license=>false, :renewed=>false})
+      @facility_1.add_service('Written Test')
+      expect(@facility_1.administer_written_test(@registrant_1)).to eq(true)
+      expect(@registrant_1.license_data).to eq({:written=>true, :license=>false, :renewed=>false})
+      expect(@registrant_2.age).to eq(16)
+      expect(@registrant_2.permit?).to eq(false)
+      expect(@facility_1.administer_written_test(@registrant_2)).to be(false)
+      @registrant_2.earn_permit
+      expect(@facility_1.administer_written_test(@registrant_2)).to be(true)
+      expect(@registrant_2.license_data).to eq({:written=>true, :license=>false, :renewed=>false})
+      expect(@registrant_3.age).to eq(15)
+      expect(@registrant_3.permit?).to eq(false)
+      expect(@facility_1.administer_written_test(@registrant_3)).to eq(false)
+      @registrant_3.earn_permit
+      expect(@facility_1.administer_written_test(@registrant_3)).to be(false)
+      expect(@registrant_3.license_data).to eq({:written=>false, :license=>false, :renewed=>false})
+    end
```

Comment:
This test is very thorough, nice job!
One suggestion I have is to break it up into multiple 'it' blocks, for better maintainability and readability in the long run. The tests would be the same, but spread out across examples that might look something like
```
it 'does not administer written test when facility does not offer the service' do
  ...
end

it 'administers a written test to an eligible registrant when the facility offers the service' do
  ...
end

it 'does not administer written test to registrants without a permit' do
  ...
end

it 'does not administer written test to registrants under 16' do
  ...
end
```

Alternatively, you could further organize your tests using more `describe` blocks.
```
describe 'facility offers written test service' do
  describe 'eligible registrants' do
    it 'administers a written test' do
      ...
    end
  end

  describe 'ineligible registrants' do
    it 'does not administer written test if registrant has no permit' do
      ...
    end

    it 'does not administer written test if registrant is under 16' do
      ...
    end
  end
end
```
And so on.